develop/blockchains/maintenance/runtime-metrics-monitoring.md

```diff
@@ -1,4 +1,111 @@
+
+[Substrate's client telemetry](https://paritytech.github.io/polkadot-sdk/master/sc_telemetry/index.html){target=\_blank} enables real-time data ingestion, which can be visualized on a client dashboard. The telemetry process uses tracing and logging to gather operational data. This data is sent through a tracing layer to a background task called the [TelemetryWorker](https://paritytech.github.io/polkadot-sdk/master/sc_telemetry/struct.TelemetryWorker.html){target=\_blank}, which then forwards it to configured remote telemetry servers.
+
+If multiple Substrate nodes run within the same process, the telemetry system uses a tracing::Span to distinguish data from each node. This ensures that each task, managed by the sc-service's [TaskManager](https://paritytech.github.io/polkadot-sdk/master/sc_service/struct.TaskManager.html){target=\_blank}, inherits a span for data consistency, making it easy to track parallel node operations.
+
+As a developer or node operator, the telemetry system handles most of the technical setup. Collected data is automatically sent to a default telemetry server, where it’s aggregated and displayed on a dashboard, making it easy to monitor network performance and identify issues.
+
+## Runtime Metrics
+
+Substrate exposes a variety of metrics about the operation of your network, such as the number of peer connections, memory usage, and block production. To capture and visualize these metrics, you can configure and use tools like Prometheus and Grafana. At a high level, Substrate exposes telemetry data that can be consumed by the Prometheus endpoint and then presented as visual information in a Grafana dashboard or graph. The provided diagram offers a simplified overview of how the interaction between Substrate, Prometheus, and Grafana can be configured to display information about node operations.
+
+```mermaid
+graph TD
+  subnode([Substrate node]) --> telemetrystream[Exposed telemetry stream]
+  telemetrystream --> prom[Prometheus]
+  configuration[Configuration: Every 1 minute] --> prom
+  prom --> endpoint[Endpoint: Every 1 minute]
+  endpoint --> grafana[Grafana]
+  grafana --> useropen[User opens a graph]
+  useropen --> getmetric[GET substrate_peers_count from timestamp X to timestamp Y]
+  prom --> localdata[Local Prometheus data]
+  localdata --> getmetric
+```
+
+The diagram shows the flow of data from the Substrate node to the monitoring and visualization components. The Substrate node exposes a telemetry stream, which is consumed by Prometheus. Prometheus is configured to collect data every minute and store it. Grafana is then used to visualize the data, allowing the user to open graphs and retrieve the `substrate_peers_count` metric from a specific time range.
```

Comment:
It might be confusing for readers to understand that `getmetric[GET substrate_peers_count from timestamp X to timestamp Y]` is an example. Maybe don't include it in the diagram or explicitly mention that it is an example